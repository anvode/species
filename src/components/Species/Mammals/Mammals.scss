.mammals {
    &__title {
        text-align: center;
    }
}
