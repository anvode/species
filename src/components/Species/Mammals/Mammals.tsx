import * as React from 'react';
import { useContext } from 'react';

import { SpeciesContext } from '../../../context/species/SpeciesContext';

import './Mammals.scss';

export interface MammalsProps {}

const Mammals: React.FC<MammalsProps> = () => {
    const { state: {mammals, region} } = useContext(SpeciesContext);

    return <div className="mammals">
        <h2 className="mammals__title">Mammals in {region}</h2>
        {mammals.length > 0 ? (
            <div className="species-results">
                <p>{mammals.length} Species found</p>

                {mammals.map(item => (
                    <div key={item.taxonid} className="species-results__item">
                        <div className={'species-results__name'}>{item.scientific_name}</div>
                        <div className="species-results__value">{item.class_name}</div>
                    </div>
                ))}
            </div>
        ): 'No results'}
    </div>;
};

export default Mammals;