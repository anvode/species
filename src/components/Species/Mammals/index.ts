import Mammals from './Mammals';

export default Mammals;