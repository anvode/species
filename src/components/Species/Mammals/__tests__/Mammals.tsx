import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import Mammals from '../Mammals';

afterEach(cleanup);

describe('Mammals', () => {
    it('snapshot Mammals', () => {
        const { asFragment } = render(<Mammals />);
        expect(asFragment()).toMatchSnapshot();
    });
});

