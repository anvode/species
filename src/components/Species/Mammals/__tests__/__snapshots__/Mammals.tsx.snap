// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mammals snapshot Mammals 1`] = `
<DocumentFragment>
  <div
    class="mammals"
  >
    <h2
      class="mammals__title"
    >
      Mammals in 
    </h2>
    No results
  </div>
</DocumentFragment>
`;
